src/SFA.DAS.Assessor.Functions/Functions/Ofqual/OfqualDownloader.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.Assessor.Functions.Domain.Entities.Ofqual;
+using SFA.DAS.Assessor.Functions.Domain.OfqualImport.Interfaces;
+using SFA.DAS.Assessor.Functions.Infrastructure;
+
+namespace SFA.DAS.Assessor.Functions.Functions.Ofqual
+{
+    internal abstract class OfqualDownloader
+    {
+        private readonly IOfqualDownloadsBlobFileTransferClient _blobFileTransferClient;
+        private readonly HttpClient _httpClient;
+        private readonly OfqualDataType _ofqualFileType;
+        private readonly string _ofqualDataTypeKeyword;
+
+        protected OfqualDownloader(
+            IOfqualDownloadsBlobFileTransferClient blobFileTransferClient,
+            HttpClient httpClient,
+            OfqualDataType ofqualFileType)
+        {
+            _blobFileTransferClient = blobFileTransferClient;
+            _httpClient = httpClient;
+            _ofqualFileType = ofqualFileType;
+            _ofqualDataTypeKeyword = GetOfqualDataTypeKeyword(ofqualFileType);
+        }
+
+        protected async Task<string> DownloadData(ILogger logger)
+        {
+            logger.LogInformation("Downloading {datatype} data from {address}.", _ofqualDataTypeKeyword, _httpClient.BaseAddress);
+
+            try
+            {
+                using HttpResponseMessage response = await _httpClient.GetAsync(_httpClient.BaseAddress);
```

Comment:
_httpClient.GetAsync(string.Empty) would be better rather than forcing the absolute address which is already stored in the _httpClient instance

Reply:
Changed